src/components/builder/i18n.stories.tsx

```diff
@@ -21,83 +21,91 @@ export default {
-  // FIXME: proper typing with nested keys
-  useManageTranslations(translatableFields as any);
+const Body: React.FC<BodyProps> = ({fieldLabels}) => {
   return (
     <>
-      <TextField name="translatableField" label="Translatable field" />
+      <TextField name="label" label="Label" />
+      <TextField name="description" label="Description" />
       <TextField name="nonTranslatableField" label="Non-translatable field" />
-      <TextField name="nested.translatableField" label="Nested translatable field" />
-      <ComponentTranslations />
+      <ComponentTranslations<DummyComponent> propertyLabels={fieldLabels} />
     </>
   );
 };
 
-interface StoryArgs {
-  translatableFields: string[];
+interface StoryArgs extends BodyProps {
   initialValues: {
     [key: string]: any;
   };
 }
 
-const Template: StoryFn<React.FC<StoryArgs>> = ({translatableFields, initialValues}) => (
+type Story = StoryObj<StoryArgs>;
+
+const render: StoryFn<React.FC<StoryArgs>> = ({fieldLabels, initialValues}) => (
   <Formik enableReinitialize initialValues={initialValues} onSubmit={console.log}>
-    <Body translatableFields={translatableFields} />
+    <Body fieldLabels={fieldLabels} />
   </Formik>
 );
 
-export const Default = {
-  render: Template,
+export const Default: Story = {
+  render: render,
 
   play: async ({canvasElement}) => {
     const canvas = within(canvasElement);
 
-    // Translations component should initialize/synchronize with available literals and
-    // translations from the store.
+    // Translations component must display the registered/existing translations.
+    const translationField1 = canvas.getByLabelText('Translation for "label"');
+    expect(translationField1).toBeVisible();
+
+    const translationField2 = canvas.queryByLabelText('Translation for "nonTranslatableField"');
+    expect(translationField2).toBeNull();
+
     await waitFor(async () => {
-      const literal1Inputs = canvas.queryAllByDisplayValue('Literal 1');
-      expect(literal1Inputs).toHaveLength(2);
-      await canvas.findByDisplayValue('Vertaling 1');
+      const literal1Reference = canvas.getByText('Hi there {{ firstName }}');
+      expect(literal1Reference).toBeVisible();
+      await canvas.findByDisplayValue('Hallo daar, {{ firstName }}');
     });
 
     // Enter a value in the non-translatable field
     await userEvent.type(canvas.getByLabelText('Non-translatable field'), 'Literal 2');
-    await waitFor(async () => {
-      const literal2Inputs = canvas.queryAllByDisplayValue('Literal 2');
-      expect(literal2Inputs).toHaveLength(1);
-    });
-
-    // Enter a value for the nested translatable field
-    await userEvent.type(canvas.getByLabelText('Nested translatable field'), 'Literal 3');
-    await waitFor(async () => {
-      const literal3Inputs = canvas.queryAllByDisplayValue('Literal 3');
-      expect(literal3Inputs).toHaveLength(2);
-    });
+    expect(translationField2).toBeNull();
```

Comment:
This re-evaluates the query?

Reply:
What was wrong with the previous interactions? I thought we only change where translations are stored. So I kinda only expect changes in the setup of the canvas.

Reply:
> This re-evaluates the query?

yeah you can re-use references, in playwright too it uses a locator expression which when combined with the `expect` only then "evaluates" the locator

Reply:
> What was wrong with the previous interactions? I thought we only change where translations are stored. So I kinda only expect changes in the setup of the canvas.

The previous setup was a table mapping from literal -> translation without any accessible labels. Now the table has three columns with proper labels, a reference for the value currently defined in the form and a field for the translation.